libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/UIDevice+SFHardware.m

```diff
@@ -187,119 +216,129 @@ - (NSNumber *)sfsdk_freeDiskSpace {
-    if ([platform isEqualToString:@"iPad3,5"])          return UIDevice4GiPad;
-    if ([platform isEqualToString:@"iPad3,6"])          return UIDevice4GiPad;
-    if ([platform isEqualToString:@"iPad4,1"] ||
-        [platform isEqualToString:@"iPad4,2"] ||
-        [platform isEqualToString:@"iPad4,3"])          return UIDevice1GiPadAir;
-    if ([platform isEqualToString:@"iPad4,4"] ||
-        [platform isEqualToString:@"iPad4,5"] ||
-        [platform isEqualToString:@"iPad4,6"])          return UIDevice2GiPadMini;
-    if ([platform isEqualToString:@"iPad4,7"] ||
-        [platform isEqualToString:@"iPad4,8"] ||
-        [platform isEqualToString:@"iPad4,9"])          return UIDevice3GiPadMini;
-    if ([platform isEqualToString:@"iPad5,1"] ||
-        [platform isEqualToString:@"iPad5,2"])          return UIDevice4GiPadMini;
-    if ([platform isEqualToString:@"iPad5,3"] ||
-        [platform isEqualToString:@"iPad5,4"])          return UIDevice2GiPadAir;
-    if ([platform isEqualToString:@"iPad6,3"] ||
-        [platform isEqualToString:@"iPad6,4"])          return UIDevice97InchiPadPro;
-    if ([platform isEqualToString:@"iPad6,7"] ||
-        [platform isEqualToString:@"iPad6,8"])          return UIDevice129InchiPadPro;
-    if ([platform isEqualToString:@"iPad6,11"] ||
-        [platform isEqualToString:@"iPad6,12"])          return UIDevice5GiPad;
-    if ([platform isEqualToString:@"iPad7,1"] ||
-        [platform isEqualToString:@"iPad7,2"])          return UIDevice2G129InchiPadPro;
-    if ([platform isEqualToString:@"iPad7,3"] ||
-        [platform isEqualToString:@"iPad7,4"])          return UIDevice105InchIpadPro;
-    if ([platform isEqualToString:@"iPad7,5"] ||
-        [platform isEqualToString:@"iPad7,6"])          return UIDevice6GiPad;
-    if ([platform isEqualToString:@"iPad8,1"] ||
-        [platform isEqualToString:@"iPad8,2"] ||
-        [platform isEqualToString:@"iPad8,3"] ||
-        [platform isEqualToString:@"iPad8,4"])          return UIDevice11InchIpadPro;
-    if ([platform isEqualToString:@"iPad8,5"] ||
-        [platform isEqualToString:@"iPad8,6"] ||
-        [platform isEqualToString:@"iPad8,7"] ||
-        [platform isEqualToString:@"iPad8,8"])          return UIDevice3G129InchiPadPro;
+    // Placeholder for unknown platforms or iFPGA (rare and mostly unused)
+    if ([platform isEqualToString:@"iFPGA"]) return UIDeviceIFPGA;
     
+    // Simulators
+#if !TARGET_OS_VISION
+    if ([self sfsdk_isSimulator]) {
+        UIDevice *device = [UIDevice currentDevice];
+        
+        switch (device.userInterfaceIdiom) {
+            case UIUserInterfaceIdiomPhone:
+                return UIDeviceSimulatoriPhone;
+            case UIUserInterfaceIdiomPad:
+                return UIDeviceSimulatoriPad;
+            default:
+                return UIDeviceSimulator; // Default case for unknown idioms (e.g., visionOS or future devices)
```

Comment:
I still think `#if !TARGET_OS_VISION` should be removed but if not the comment is confusing because visionOS will never execute here while that check in place